content/2. Ondersteunende-informatie/3. Processen/3. Ontwerpproces/1. Opstellen ontwerp/Schematechnieken/1. Klassendiagram.md

```diff
@@ -0,0 +1,163 @@
+---
+title:
+  - Het klassendiagram
+tags:
+  - UML
+  - Klassendiagram
+difficulty: 2
+taxonomie:
+  - oo-15.1.Klassendiagram.OI
+  - oo-15.2.Klassendiagram.OI
+draft: true
+---
+
+## Wat is een klassendiagram
+
+Het klassendiagram is een ontwerptechniek voor het modelleren van de zogenaamde `structuur` van een softwaresysteem. Het bevat informatie over de `eigenschappen`(attributen) en het `gedrag` (methoden) van de klassen in het systeem. Ook bevat het diagram informatie over de `samenhang` (relaties) tussen de diverse klassen. 
+
+Onderstaand diagram is een voorbeeld van een klassendiagram. Hierin zie je de structuur van een eenvoudig hotel softwaresysteem. Dit voorbeeld is netjes uitwerkt maar kan zijn ontstaan terwijl je met je collega's bij een whiteboard stond.
+
+```mermaid
+classDiagram
+
+class Hotel {
+    +String naam
+    +String locatie
+    +geefKamers(bedrag)
+}
+
+
+class Kamer {
+	+String kamerNummer
+	+int aantalBedden
+	+decimal prijsPerNacht
+}
+
+Hotel --> Kamer
+```
+*Diagram 1: een eenvoudig voorbeeld van een klassendiagram*
+
+Onderstaande foto is het resultaat van een `brainstorm` op een whiteboard. Een nuttige toepassing als je nog in de ideeën fase bent.
+![[OI_Klassendiagram_1.jpg]]
+*Figuur 1: Een voorbeeld van een uitwerking op het whiteboard*
+## Hoe zit het klassendiagram in elkaar
+
+Klassendiagrammen kun je tegenkomen in een `functioneel ontwerp` en in een `technisch ontwerp`. In een functioneel ontwerp komt geen techniek voor. Je opdrachtgever of eindgebruiker zal een klassendiagram dat in een functioneel ontwerp is opgenomen kunnen begrijpen. Het beschrijft het domein waarvoor we een applicatie gaan maken. We noemen een dergelijk klassendiagram dan ook een `domeinmodel`.
```

Comment:
"Je opdrachtgever of eindgebruiker zal een klassendiagram dat in een functioneel ontwerp is opgenomen **moeten** kunnen begrijpen."
```suggestion
Klassendiagrammen kun je tegenkomen in een `functioneel ontwerp` en in een `technisch ontwerp`. In een functioneel ontwerp komt geen techniek voor. Je opdrachtgever of eindgebruiker zal een klassendiagram dat in een functioneel ontwerp is opgenomen moeten kunnen begrijpen. Het beschrijft het domein waarvoor we een applicatie gaan maken. We noemen een dergelijk klassendiagram dan ook een `domeinmodel`.
```